internal/gitinterface/commit.go

```diff
@@ -0,0 +1,127 @@
+package gitinterface
+
+import (
+	"errors"
+	"io"
+	"os/exec"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+var ErrUnableToSign = errors.New("unable to sign Git object")
+
+func Commit(repo *git.Repository, treeHash plumbing.Hash, targetRef string,
+	message string, sign bool) error {
+	gitConfig, err := repo.ConfigScoped(config.GlobalScope)
+	if err != nil {
+		return err
+	}
+
+	curRef, err := repo.Reference(plumbing.ReferenceName(targetRef), true)
+	if err != nil {
+		return err
+	}
+
+	author := object.Signature{
+		Name:  gitConfig.User.Name,
+		Email: gitConfig.User.Email,
+		When:  time.Now(),
```

Comment:
if you swap this with something like github.com/jonboulle/clockwork, you can deterministically create commits, which makes testing soooooo much easier since you can assert commit SHAs.

Reply:
This is fantastic, thanks!

Reply:
Added, I moved the commit object creation to a helper which we can test independently for deterministic commits. Thanks for the tip!